lib/library_web/controllers/login_controller.ex

```diff
@@ -0,0 +1,159 @@
+defmodule LibraryWeb.LoginController do
+  use LibraryWeb, :controller
+  alias Library.Users
+  @elixir_auth_github Application.get_env(:library, :elixir_auth_github) || ElixirAuthGithub
+
+  @httpoison Application.get_env(:library, :httpoison) || HTTPoison
+
+  @doc """
+    Just shows the user the plain index.html
+  """
+  def index(conn, _params) do
+    render conn, "index.html"
+  end
+
+
+  @doc """
+    Creates a github URL with the scope for getting the user's email and orgs
+    renders the index.html if there is a failure constructing it (can only
+    happen when elixir_auth_github is not configured correctly/env variables
+    are not set), and redirects to github login if the url is correct
+  """
+  def login(conn, _params) do
+
+    case @elixir_auth_github.login_url_with_scope(["user:email", "read:org"]) do
+      {:err, reason} ->
```

Comment:
If we aren't going to use the reason somewhere below here, we can put it as `_reason` so we don't have unused variable warnings.